### PR TITLE
Disable idleEndpointCheck if set to 0

### PR DESCRIPTION
**Pull request by xinlian12**
Disable idleCheckpointCheck if IdleEndpointTimeout = 0. Relates to Issue https://github.com/Azure/azure-sdk-for-java/issues/23067

### CODE REVIEW

**Review by FabianMeiswinkel — APPROVED** [review]
Thx

**Review by moderakh — APPROVED** [review]
LGTM. thanks

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java [hunk 1]
```diff
@@ -975,6 +975,14 @@ private void newTimeout(
         final long idleEndpointTimeoutInNanos,
         final long requestTimerResolutionInNanos) {
 
+        if (idleEndpointTimeoutInNanos == 0) {
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java]: Based on the offline discussion with @xinlian12 , this change is not correct. We are working to rectify it. Blocking the PR until it is fixed and we run the full CI to make sure there is no regression.

**Review by kushagraThapar — CHANGES_REQUESTED** [review]

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java (hunk 1 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java]: Thanks for capture it. This method also contains acquisition queue check. Updated

**Review by kushagraThapar — APPROVED** [review]

**Comment by xinlian12** [comment]
Tested following tests locally and passed:
createItem_withBulkAndThroughputControl
queryWithFeedRangeFiltering
readUserDefinedFunctions
loadCachesAndOpenConnectionsToServiceSyncContainer
populateDocuments
sessionNotAvailableRetrySingleMaster

**Pull request merged**